microprofile/opentracing-smallrye/src/main/java/org/wildfly/microprofile/opentracing/smallrye/TracingCDIExtension.java

```diff
@@ -42,7 +45,18 @@ public static void registerApplicationTracer(ClassLoader classLoader, Tracer tra
 
     public void registerTracerBean(@Observes AfterBeanDiscovery abd) {
         abd.addBean().addTransitiveTypeClosure(Tracer.class).produceWith(i -> {
-            return TRACERS.get(Thread.currentThread().getContextClassLoader());
+            ClassLoader loader;
```

Comment:
This entire block can be removed and instead just call WildFlySecurityManager.getCurrentContextClassLoaderPrivileged()